### PR TITLE
Two ultrasonic sensor example for sentry

### PR DESCRIPTION
**Pull request by AlchemicRonin**

### CODE REVIEW

File: boards/DJI_Board_TypeA/Core/Src/gpio.c [hunk 1]
```diff
@@ -129,10 +141,18 @@ void MX_GPIO_Init(void)
   HAL_GPIO_Init(LED_RED_GPIO_Port, &GPIO_InitStruct);
 
   /*Configure GPIO pin : PtPin */
-  GPIO_InitStruct.Pin = SD_EXTI_Pin;
-  GPIO_InitStruct.Mode = GPIO_MODE_INPUT;
+  GPIO_InitStruct.Pin = ULTRASONIC1_Trig_Pin;
```
Comment (Fred-Hou) [inline boards/DJI_Board_TypeA/Core/Src/gpio.c:144]: change the name of the pin

File: boards/DJI_Board_TypeA/Core/Src/gpio.c [hunk 2]
```diff
@@ -129,10 +141,18 @@ void MX_GPIO_Init(void)
...
+  GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_PP;
+  GPIO_InitStruct.Pull = GPIO_NOPULL;
+  GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
+  HAL_GPIO_Init(ULTRASONIC1_Trig_GPIO_Port, &GPIO_InitStruct);
+
+  /*Configure GPIO pin : PtPin */
+  GPIO_InitStruct.Pin = ULTRASONIC2_Trig_Pin;
```
Comment (Fred-Hou) [inline boards/DJI_Board_TypeA/Core/Src/gpio.c:151]: change the name of the pin

File: examples/ultrasonic/main.cc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+void RM_RTOS_Init(void) {
+	print_use_uart(&huart8);
+  HAL_TIM_Base_Start_IT(&htim2);
+  ultrasonic1 = new bsp::Ultrasonic(ULTRASONIC1_Trig_GPIO_Port,
+                                   ULTRASONIC1_Trig_Pin,
+                                   ULTRASONIC1_Echo_GPIO_Port,
+                                   ULTRASONIC1_Echo_Pin,
+                                   TIM2);
+	ultrasonic2 = new bsp::Ultrasonic(ULTRASONIC2_Trig_GPIO_Port,
```
Comment (Fred-Hou) [inline examples/ultrasonic/main.cc:39]: extra whitespace

File: examples/ultrasonic/main.cc [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  while (true) {
+    float distance1 = ultrasonic1->GetDistance();
+		float distance2 = ultrasonic2->GetDistance();
+    set_cursor(0, 0);
+    clear_screen();
+    if (distance1 > 0 && distance2 > 0) {
+      print("Distance1: %.2f cm   Distance2:  %.2f cm\r\n", distance1, distance2);
+    } else if (distance1 > 0 && distance2 < 0) {
+	    print("Distance1: %.2f cm   ERROR!!!\r\n", distance1);
+		} else if (distance2 > 0 && distance1 < 0) {
```
Comment (Fred-Hou) [inline examples/ultrasonic/main.cc:58]: extra whitespace

**Review by Fred-Hou — CHANGES_REQUESTED** [review]

File: boards/DJI_Board_TypeA/Middlewares/ST/STM32_USB_Device_Library/Class/CDC/Src/usbd_cdc.c [hunk 5]
```diff
@@ -446,8 +446,6 @@ __ALIGN_BEGIN static uint8_t USBD_CDC_OtherSpeedCfgDesc[USB_CDC_CONFIG_DESC_SIZ]
   * @{
   */
```
Comment (0-EricZhou-0) [inline boards/DJI_Board_TypeA/Middlewares/ST/STM32_USB_Device_Library/Class/CDC/Src/usbd_cdc.c:448]: This part is specifically modified by Alvin to resolve USB bug presented in Middleware. This file should not be changed.

File: shared/bsp/bsp_ultrasonic.cc [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+ *                                                                          *
+ *  You should have received a copy of the GNU General Public License       *
+ *  along with this program. If not, see <http://www.gnu.org/licenses/>.    *
+ *                                                                          *
+ ****************************************************************************/
+
+#include "bsp_ultrasonic.h"
+
+namespace bsp {
+
```
Comment (0-EricZhou-0) [inline shared/bsp/bsp_ultrasonic.cc:24]: Code inside namespace should not be indented according to other files inside the repo.

File: shared/bsp/bsp_ultrasonic.h [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+ *                                                                          *
+ ****************************************************************************/
+
+#pragma once
+
+#include "bsp_gpio.h"
+#include "main.h"
+
+namespace bsp {
+
```
Comment (0-EricZhou-0) [inline shared/bsp/bsp_ultrasonic.h:27]: Same indentation problem

**Review by 0-EricZhou-0 — COMMENTED** [review]
Pin assignments should not be hard coded to any specific name (e.g. ultrasonic_xxx here)
Also please change indentation.

**Pull request closed without merging**